### DISCUSSION TITLE
Remaining features / tasks before public release

### DISCUSSION THREAD

**Discussion by Nexucis** [perses/perses#875] [General]
This discussion is about what needs to be done before announcing Perses can be used for the first time publicly (even if it's not prod ready).

Here a first list : 

- Dashboard CRUD : https://github.com/perses/perses/issues/823 https://github.com/perses/perses/issues/818
- Project CRUD
- https://github.com/perses/perses/issues/806
- Bugs reported in https://docs.google.com/spreadsheets/d/1neE7hzy3wx57u6swvZnHbN7PVeuxnx2cBQqSDNNuTHk/edit with P1 and P2 priority
- if we have time, nested folder management https://github.com/perses/perses/issues/183

Initial target date to announce Perses: End of February

**Comment by Nexucis** [comment]
I think we need to finish also the discussion about the global variable: https://github.com/perses/perses/discussions/226

**Comment by juliepagano** [comment]
Below is a collection of items from the UI devs on the chronosphere side with links to relevant issues, where I could find them.
- Set up storybook to provide better example and documentation for consumers. https://github.com/perses/perses/issues/918 -- Note from 2023-01-23 meeting: this may not be critical if we're focusing on the app use case
- Improved test coverage to increase confidence in the stability of the project
- UI package organizational improvements and associated improvements to the contributor documentation for the UI sections of the code.
- Set up a play/demo site with reliable test data. https://github.com/perses/perses/issues/883, https://github.com/perses/perses/issues/919
- Improved APIs for the dashboard header components.
- Finish discussion on [[BREAKINGCHANGE] Add a consistent JSON tab for every panel plugin](https://github.com/perses/perses/pull/922)

As part of putting that list together, a good question came up around the focus of the release. This would be a good discussion topic for our next meeting.
> For the first "public" release, are we targeting Perses as a standalone application or UI framework? I think answering that will help frame what needs to be done.

**Reply by sjcobb** [reply]
It'd be good to fix #894 as well to make using our FE packages easier, I ran into some issues when trying to embed Perses that required finding workarounds ([example](https://github.com/sjcobb/perses-embed-example/blob/main/next.config.js)).

**Comment by juliepagano** [comment]
There's an open question about schema versioning that came up in the 2022-01-23 meeting.

**Comment by Nexucis** [comment]
This discussion can be closed as we made a presentation at PromCon 2023 to announce Perses in the opensource community.

Next target will be to announce it's totally ready to use at PromCon 2024.
